internal/persistence/persistors/mysqlstore/store_category_test.go

```diff
@@ -299,46 +291,62 @@ func Test_UpdateCategories_Success(t *testing.T) {
 	assert.Equal(t, paginatedCategories.Categories[0].Name+" new", cat.Name)
 }
 
-type deleteCategoryTestCase struct {
-	name       string
-	id         int
-	assertions func(t *testing.T, db *sqlx.DB, id int, err error)
-	mutations  func(t *testing.T, db *sqlx.DB)
-}
+func Test_UpdateCategories_Fail(t *testing.T) {
+	db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t)
+	defer cleanup()
 
-func getDeleteCategoryTestCases() []deleteCategoryTestCase {
-	return []deleteCategoryTestCase{
-		{
-			name: "success",
-			id:   1,
-			assertions: func(t *testing.T, db *sqlx.DB, id int, err error) {
-				require.Nil(t, err, "unexpected non-nil error")
-				entry, err := mysqlmodel.FindCategory(context.TODO(), db, id)
-				require.NoError(t, err, "unexpected error fetching the category")
+	cfg := &Config{
+		Logger:        testLogger,
+		QueryTimeouts: testQueryTimeouts,
+	}
 
-				assert.Equal(t, 0, entry.IsActive)
-			},
-			mutations: func(t *testing.T, db *sqlx.DB) {
-				entry := mysqlmodel.Category{
-					CategoryTypeRefID: 1,
-					Name:              "test",
-				}
-				err := entry.Insert(context.TODO(), db, boil.Infer())
-				assert.NoError(t, err, "unexpected insert error")
-			},
+	m, err := New(cfg)
+	require.NoError(t, err, "unexpected error")
+	require.NotNil(t, m, "unexpected nil")
+
+	txHandler, err := mysqltx.New(&mysqltx.Config{
+		Logger:       testLogger,
+		Db:           db,
+		DatabaseName: cp.Database,
+	})
+	require.NoError(t, err, "unexpected error creating the tx handler")
+
+	txHandlerDb, err := txHandler.Db(testCtx)
+	require.NoError(t, err, "unexpected error fetching the db from the tx handler")
+	require.NotNil(t, txHandlerDb, "unexpected nil tx handler db")
+
+	paginatedCategories, err := m.GetCategories(testCtx, txHandlerDb, nil)
+	require.NoError(t, err, "unexpected error fetching the categories from the database")
+	require.NotNil(t, txHandlerDb, "unexpected nil categories")
+	require.True(t, len(paginatedCategories.Categories) > 0, "unexpected empty categories")
+
+	updateCategory := model.UpdateCategory{
+		Id: paginatedCategories.Categories[1].Id,
+		CategoryTypeRefId: null.Int{
+			Int:   paginatedCategories.Categories[0].CategoryTypeRefId,
+			Valid: true,
+		},
+		Name: null.String{
+			String: paginatedCategories.Categories[0].Name,
+			Valid:  true,
```

Comment:
make all the details come from the mutation

Reply:
ahh but we talked about it coming from the migration.. okay **this is fine**!

NO changes needed